pimd/pim_neighbor.c

```diff
@@ -141,6 +142,16 @@ int pim_if_dr_election(struct interface *ifp)
 		pim_if_update_join_desired(pim_ifp);
 		pim_if_update_could_assert(ifp);
 		pim_if_update_assert_tracking_desired(ifp);
+
```

Comment:
Wouldn't this be better:
```
diff --git a/pimd/pim_neighbor.c b/pimd/pim_neighbor.c
index ca7ca1140..4b5b5472e 100644
--- a/pimd/pim_neighbor.c
+++ b/pimd/pim_neighbor.c
@@ -110,6 +110,7 @@ int pim_if_dr_election(struct interface *ifp)
 {
        struct pim_interface *pim_ifp = ifp->info;
        struct in_addr old_dr_addr;
+       bool i_am_old_dr = PIM_I_am_DR(pim_ifp);
 
        ++pim_ifp->pim_dr_election_count;
 
@@ -141,6 +142,10 @@ int pim_if_dr_election(struct interface *ifp)
                pim_if_update_join_desired(pim_ifp);
                pim_if_update_could_assert(ifp);
                pim_if_update_assert_tracking_desired(ifp);
+
+               if (!PIM_I_am_DR(pim_ifp) && i_am_old_dr)
+                       pim_reg_del_on_couldreg_fail(ifp);
+
                return 1;
        }
 
```
Then you don't need to store anything in `struct pim_iface`

Reply:
There were 2 reasons behind reason why I thought this would be better.
1. There could be a scenario where we were dr already and  our ip addr is changed and we are now non-dr. This situation, PIM_I_am_DR(pim_ifp) even at the beginning will not give us the fact that we were dr earlier.
2. Going forward was planning to replace the macro with this stored flag as we don't have to derive dr by comparison.